blueprints/to-locks.md

```diff
@@ -0,0 +1,268 @@
+
+request JSON:
+```JSON
+{
+  "cdn": "bar",
+  "message": "snapping cdn",
+  "soft": false
+}
+```
+
+response JSON:
+```JSON
+{
+  "alerts": [
+    {
+      "text": "CDN lock acquired!",
+      "level": "success"
+    }
+  ],
+  "response": {
+    "username": "foo",
+    "cdn": "bar",
+    "message": "snapping cdn",
+    "soft": false,
+    "lastUpdated": "2021-05-10T17:05:30-06:00"
+  }
+}
+```
+
+##### `DELETE /cdn_locks?cdn=bar`
+
+response JSON:
+```JSON
+{
+  "alerts": [
+    {
+      "text": "Cdn lock deleted",
+      "level": "success"
+    }
+  ],
+  "response": {
+    "username": "foo",
+    "cdn": "bar",
+    "message": "snapping cdn",
+    "soft": false,
+    "lastUpdated": "2021-05-10T17:05:30-06:00"
+  }
+}
+```
+
+#### Client Impact
+
+New Go client methods will be added for the `/cdn_locks` endpoints in order to write TO API tests for the new endpoints.
+
+#### Data Model / Database Impact
+A new database table for `cdn_lock`, as described below, will be created.
+```text
+            Table "traffic_ops.cdn_lock"
+     Column    |  Type                    | Collation | Nullable | Default
+---------------+--------------------------+-----------+----------+--------
+ id            | bigint                   |           | not null |
+ username      | text                     |           | not null |
+ cdn_name      | text                     |           | not null |
+ message       | test                     |           |          |
+ soft          | boolean                  |           | not null | true
+ last_updated  | timestamp with time zone |           | not null | now() 
+Indexes:
+    "pk_cdn_lock" PRIMARY KEY(cdn)
+Foreign-key constraints:
+    "fk_lock_cdn" FOREIGN KEY (cdn) REFERENCES cdn(name)
+    "fk_lock_username" FOREIGN KEY (username) REFERENCES tm_user(username)
+```
+
+The `capability` table will need to add another capability by the name of `delete-all-locks` and the `all-write` capability 
+will include this new capability. 
+### ORT Impact
+No impact
+
+### Traffic Monitor Impact
+No impact
+
+### Traffic Router Impact
+No impact
+
+### Traffic Stats Impact
+No impact
+
+### Traffic Vault Impact
+No impact
+
+### Documentation Impact
+All new endpoints will need to be documented, along with the documentation explaining `cdn_locks`. 
+
+### Testing Impact
+Client/API integration tests should be written to verify the
+functionality as described in [API](#sec:api).
+
+### Performance Impact
+We do not anticipate significant performance impact, as this process will just entail one additional check
+in the database before performing a snap or queue operation.
+
+### Security Impact
+We do not anticipate any impact on security due to the introduction of `cdn_locks`.
+
+### Upgrade Impact
+There will be two database migrations, one to create the `cdn_locks` table, and the other to add
+the new capability. This does not depend on any existing data, so nothing should ideally cause this migration to fail.
```

Comment:
Oh, also, you don't need a migration to add the new capabilities. Those can just go in `seeds.sql`.